src/Components/App/App.js

```diff
@@ -21,13 +23,22 @@ function App({ grabLocalCodes }) {
     }
   }
 
-  useEffect(() => {
+  useLayoutEffect(() => {
     storeMyCodes()
-  }, [])
+    let landing = window.location.pathname === '/' ?
+    'main-page' : 'not-main'
+    setLocation(landing)
+  },[])
+
+  history.listen((location) => {
```

Comment:
what's history.listen do?

Reply:
it is listening to the URL and running a function when it changes